hs/src/Reach/Eval/Core.hs

```diff
@@ -3853,6 +3853,9 @@ doToConsensus ks (ToConsensusRec {..}) = locAt slptc_at $ do
   ensure_live "to consensus"
   let st_pure = st {st_mode = SLM_ConsensusPure}
   let pdvs = st_pdvs st
+  allClasses <- and <$> traverse is_class (S.toList whos)
+  when (not (st_after_first st) && allClasses) $ do
```

Comment:
There should also be exactly one

Reply:
So, `Anybody.publish()` and `race(Alice, Bob).publish()` are no longer legal first publications?

Reply:
Correct, because that means that both `Alice` and `Bob` would both try to `deploy` and so they'd both create a new contract and not be able to share the contract info with each other

Reply:
(It is literally possible to allow it, but I don't think the typical frontend would be correct with this backend.)